hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -4570,7 +4570,7 @@ func (r *HostedClusterReconciler) reconcileKubevirtPlatformDefaultSettings(ctx c
 	// auto generate the basedomain by retrieving the default ingress *.apps dns.
 	if hc.Spec.Platform.Kubevirt.BaseDomainPassthrough != nil && *hc.Spec.Platform.Kubevirt.BaseDomainPassthrough {
 		if hc.Spec.DNS.BaseDomain == "" {
-			kvInfraCluster, err := r.KubevirtInfraClients.DiscoverKubevirtClusterClient(ctx, r.Client, hc.Spec.InfraID, hc, hc.Namespace)
+			kvInfraCluster, err := r.KubevirtInfraClients.DiscoverKubevirtClusterClient(ctx, r.Client, hc.Spec.InfraID, hc.Spec.Platform.Kubevirt.Credentials, hc.Namespace, hc.Namespace)
```

Comment:
@davidvossel - I did it originally, but the test was constantly failed. Also, this is a change from the code before this PR. @orenc1 - WDYT?

Reply:
the 5th argument is used in `DiscoverKubevirtClusterClient()` only in case of local infra (otherwise the namespace used is being taken from `hc.Spec.Platform.Kubevirt.Credentials.InfraNamespace`.

since this namespace is used only for the temp dummy route for the purpose of resolving the base domain, i would say it doesn't matter much if it's created in hosted cluster or HCP namespace in case of local infra.

however, since it is running on hypershift-operator, and not on control-plane-operator, there can be a chance that HCP and HCP namespace were not yet created at that point, so it would be safer to use the HC namespace, which must exist prior of the creation of the HostedCluster.